systemtests/scripts/wait_until_up.sh

```diff
@@ -23,26 +23,30 @@ fi
 info "Expected pods: ${EXPECTED_PODS}"
 
 function waitingContainersReady {
-    ADDR_SPACE=$1
+    NAMESPACE=$1
     pods_id=$($CMD get pods -n ${NAMESPACE} | awk 'NR >1 {print $1}')
     for pod_id in ${pods_id}
     do
-        ready=$($CMD get -o json pod -n ${ADDR_SPACE}  ${pod_id} -o jsonpath={.status.containerStatuses[0].ready})
+        ready=$($CMD get -o json pod -n ${NAMESPACE}  ${pod_id} -o jsonpath={.status.containerStatuses[0].ready})
         if [[ "${UPGRADED}" == "true" ]]; then
             image=$($CMD get pod ${pod_id} -o jsonpath={.spec.containers[*].image})
-            upgraded=$(is_upgraded ${image})
-            if [[ "${upgraded}" == "true" ]]; then
-                info "Pod ${pod_id} is upgraded to ${image}"
+            initImage=$($CMD get pod ${pod_id} -o jsonpath={.spec.initContainers[*].image})
+            imageUpgraded=$(is_upgraded ${image})
+            initImageUpgraded=$(is_upgraded ${initImage})
+            if [[ "${imageUpgraded}" == "true" ]] && [[ "${initImageUpgraded}" == "true" ]]; then
+                info "Pod ${pod_id} is upgraded to ${image}, initImage: ${initImage}"
             else
-                info "Pod ${pod_id} is not upgraded, current image: ${image}"
+                info "Pod ${pod_id} is not upgraded, current image: ${image}, initImage: ${initImage}"
             fi
         else
-            upgraded="true"
+            imageUpgraded="true"
         fi
-        if [[ "${ready}" == "false" ]] || [[ "${upgraded}" == "false" ]]
+        if [[ "${ready}" == "false" ]] || [[ "${imageUpgraded}" == "false" ]] && [[ "${initImageUpgraded}" == "false" ]]
```

Comment:
Shouldn't this be a || b || c and not a || b && c ?

Reply:
yes, it should, will fix it